src/main/java/seedu/address/model/todolist/Task.java

```diff
@@ -0,0 +1,135 @@
+package seedu.address.model.todolist;
+
+/**
+ * Represents a Task in the todo list.
+ * Guarantees: details are present and not null, field values are validated, immutable.
+ */
```

Comment:
Will users be able to create Tasks without a type, priority or date? Perhaps allowing users to omit the addition of certain Task fields would increase flexibility?

Reply:
Yes, I think we should do that. Will try to change in the next PR.